test/beast/core/buffers_adaptor.cpp

```diff
@@ -85,6 +113,69 @@ class buffers_adaptor_test : public unit_test::suite
         read_size(ba, 1024);
     }
 
+    template<bool isMutable>
+    void
+    testSubrange()
+    {
+        auto exemplar = std::string("the quick brown fox jumps over the lazy dog");
+
+        auto iterate_test = [&](
+            std::size_t a,
+            std::size_t b,
+            std::size_t c)
+        {
+            static const auto func = "iterate_test";
+            auto buffers = std::vector<net::mutable_buffer>();
+            if (a)
+                buffers.push_back(net::buffer(&exemplar[0], a));
+            if (b - a)
+                buffers.push_back(net::buffer(&exemplar[a], (b - a)));
+            if (c - b)
+                buffers.push_back(net::buffer(&exemplar[b], (c - b)));
+            auto adapter = buffers_adaptor<std::vector<net::mutable_buffer>>(buffers);
+
+
+            using value_type =
+            typename std::conditional<
+                isMutable,
+                net::mutable_buffer,
+                net::const_buffer>::type;
+
+            using maybe_mutable =
+            typename std::conditional<
+                isMutable,
+                buffers_adaptor<std::vector<net::mutable_buffer>>&,
+                buffers_adaptor<std::vector<net::mutable_buffer>> const&>::type;
+
+            auto sub = buffers_adaptor_test_hook::make_subrange(static_cast<maybe_mutable>(adapter));
+            BEAST_EXPECTS(typeid(typename decltype(sub)::value_type) == typeid(value_type), func);
+
+            auto as_string = [](decltype(sub) const &sr) {
```

Comment:
What's wrong with calling `beast::buffers_to_string`? It is even more efficient than this function (since it reserves the space first)
https://www.boost.org/doc/libs/develop/libs/beast/doc/html/beast/ref/boost__beast__buffers_to_string.html